scripts/clear-signed-agreements.py

```diff
@@ -46,22 +48,41 @@ def main(stage, framework_slug, api_token, user):
                     extra={'supplier_id': supplier['supplierId']})
         api_client.unset_framework_agreement_returned(supplier['supplierId'], framework_slug, user)
 
-    signed_agreements = filter(
-        lambda x: re.search(r'/(\d+)-signed-framework-agreement.pdf', x['path']),
-        agreements_bucket.list('{}/agreements/'.format(framework_slug))
-    )
+    signed_agreements = filter_agreements(supplier_ids)
 
     for document in signed_agreements:
         logger.info("Deleting {path}", extra={'path': document['path']})
         agreements_bucket.delete_key(document['path'])
 
 
+def filter_agreements(supplier_ids=None):
+    if supplier_ids:
+        matcher = lambda x: x['path'] in ['{}-signed-framework-agreement.pdf'.format(id) for id in supplier_ids]
+    else:
+        lambda x: re.search(r'/(\d+)-signed-framework-agreement.pdf', x['path'])
+
+    return filter(
+        matcher,
+        agreements_bucket.list('{}/agreements/'.format(framework_slug))
+    )
+
+
+def get_supplier_ids_from_csv(csv_path):
+    with open(csv_path, 'r') as csvfile:
+        for row in csv.reader(csvfile):
+            # csv should only have one row
```

Comment:
Should it check that there aren't more?  Maybe fail with a message if there's more than one row?  Deleting signed agreements feels pretty hardcore, so people need to be sure what they're doing.  If they have the wrong format file then they probably are deleting stuff they shouldn't be.